components/homme/src/theta-l_kokkos/cxx/Diagnostics.hpp

```diff
@@ -35,10 +35,34 @@ class Diagnostics
     ExecViewUnmanaged<Scalar *[NP][NP][NUM_LEV_P]>  dpnh_dp_i;
   };
 
+
+  template <typename FunctorTag>
```

Comment:
Did this workaround end up being needed? I'm concerned about reducing the number of threads used in these routines in an optimized build.

Reply:
According to https://github.com/E3SM-Project/E3SM/issues/5069 , yes, for builds with asserts.

Reply:
Since that isn't an official build and we don't want to lose performance in a kernel, I recommend leaving up the issue with a note about this workaround and removing the workaround from this PR.

Reply:
Should i try to use the workaround with #if ! defined NDEBUG ?

Reply:
Oh -- i missed that you recommended NDEBUG already, thanks!

Reply:
This needs to be revisited once a bug in diagnostics is found.